spec/controllers/answers_controller_spec.rb

```diff
@@ -47,23 +47,106 @@
+        delete :destroy, id: answer, question_id: question, format: :js
+        expect(response).to render_template :destroy
       end
     end
 
     context 'No Author try delete' do
       let!(:answer) { create(:answer, question: question)}
       it 'not delete answer from the db' do
-        expect{ delete :destroy, id: answer, question_id: question }.to_not change(Answer, :count)
+        expect{ delete :destroy, id: answer, question_id: question, format: :js }.to_not change(Answer, :count)
       end
-      it 'redirect to question path' do
-        delete :destroy, id: answer, question_id: question
-        expect(response).to redirect_to question_path(answer.question)
+      it 'render template' do
+        delete :destroy, id: answer, question_id: question, format: :js
+        expect(response).to render_template :destroy
       end
     end
   end
+
+  describe 'PATCH #update' do
+    sign_in_user
+    context 'Author try update' do
+      let!(:answer) { create(:answer, question: question, user: @user) }
+      it 'assings the requested answer to @answer' do
+        patch :update, id: answer, question_id: question, answer: attributes_for(:answer), format: :js
+        expect(assigns(:answer)).to eq answer
+      end
+
+      it 'assigns th question' do
+        patch :update, id: answer, question_id: question, answer: attributes_for(:answer), format: :js
+        expect(assigns(:question)).to eq question
+      end
+
+      it 'changes answer attributes' do
+        patch :update, id: answer, question_id: question, answer: { body: 'new body'}, format: :js
+        answer.reload
+        expect(answer.body).to eq 'new body'
+      end
+
+      it 'render update template' do
+        patch :update, id: answer, question_id: question, answer: attributes_for(:answer), format: :js
+        expect(response).to render_template :update
+      end
+    end
+
+    context 'No Author try update' do
+      let!(:answer) { create(:answer, question: question)}
+      it 'not cahnge answer' do
+        patch :update, id: answer, question_id: question, answer: { body: 'new body'}, format: :js
+        answer.reload
+        expect(answer.body).to eq answer.body
+      end
+    end
+  end
+
+  describe 'PATCH #select_best_answer' do
+    let(:prev_answer){ create(:answer, best_answer: true, question: question) }
+    let(:answer){ create(:answer, question: question) }
+    
+    
+    context 'Author try select best answer' do
+      sign_in_user
+      let(:question){ create(:question, user: @user) }
+
+      before { patch :select_best_answer, id: answer }
+
+      it 'assign to requiest question' do
+        expect(assigns(:question)).to eq question
+      end
+
+      it 'assign to requiest answer' do
+        expect(assigns(:answer)).to eq answer
+      end
+
+      it 'change attribute for answer and prev-answer' do
+        answer.reload
+        prev_answer.reload
+
+        expect(answer.best_answer).to eq true
```

Comment:
`expect(answer).to be_best_answer`
` expect(prev_answer).to_not be_best_answer`